pkg/platform/abstract/platform.go

```diff
@@ -1106,3 +1150,23 @@ func (ap *Platform) getBaseImagesOverrides() map[string]string {
 func (ap *Platform) getOnbuildImagesOverrides() map[string]string {
 	return ap.Config.ImageRegistryOverrides.OnbuildImageRegistries
 }
+
+func (ap *Platform) transformDisplayNameToName(projectConfig *platform.ProjectConfig) {
+	ap.Logger.WarnWith("Transforming display name",
+		"displayName", projectConfig.Spec.DisplayName,
+		"name", projectConfig.Meta.Name)
+
+	// trim spaces
+	displayName := strings.TrimSpace(projectConfig.Spec.DisplayName)
+
+	// lower case
+	displayName = strings.ToLower(displayName)
+
+	// no spaces
+	displayName = strings.ReplaceAll(displayName, " ", "-")
+
+	// no underscores
+	displayName = strings.ReplaceAll(displayName, "_", "-")
+
+	projectConfig.Meta.Name = displayName
```

Comment:
if this assigned the name, why doesn't it also cleans up the displayName attribute as well?

Reply:
its possible - done